pkg/auth/multiauthenticator.go

```diff
@@ -5,64 +5,122 @@ import (
 	"fmt"
 )
 
-type multiAuthenticatorImpl struct {
-	authenticators map[string][]Authenticator
+// This file provides 2 implementations of Athenticator interface. Each implementation has multiple
```

Comment:
nit: `Athenticator` seems to show up here and in the PR comment. Maybe want to look for it everywhere and replace with `Authenticator`